prime-router/docs/features/0005-managing-translation-schemas.md

```diff
@@ -0,0 +1,765 @@
+    - Fallback to the existing behavior if it is not a valid URI (i.e. we're resolving an old translation schema with relative paths)
+2. Update all the translation schema references (i.e. `extends`, `schemaRef`, `schemaName`, etc.) in the existing translation schemas and settings to reference absolute paths as URLs
+    - `schema: ../common/patient` -> `schema: file:///{BASE_DIR}/metadata/hl7_mapping/common/patient.yml`
+3. Update translation schemas to reference azure blobs (i.e `azure:///{AZURE_STORAGE_LOCATION}/metadata/hl7_mapping/common/patient.yml`) and upload translation schemas to azure blob service
+4. Update all UP sender and receivers to reference the azure translation schema
+
+During the migration process, it would make sense to all take another look at the directory structure and names to make sure they make sense moving forward.
+
+#### HL7 -> FHIR
+Long term, it would be great to add support to this library (it's open sourced) that would support reading files from different file-like storage solutions rather than just disk, but this would likely not be feasible to get done in a quick enough timeframe.  Instead, before instantiating the converter, the application will sync the azure storage to a spot on the disk the library will read from.
+
+#### Cache
+**This could potentially be deferred until it's been identified that the network requests to azure blob storage are the bottleneck**
+
+Since the pipeline will now require reading several files out of the blob store at multiple spots, it will likely make sense to cache the resolved translation schemas since they will change relatively infrequently.  This could initially be done with a simple in-memory cache using the translation schema URI as the cache key.
+
+#### Pros
+- Maintains the current file based model for the translation schemas
+- Translation schemas can be easily edited by hand via the Azure UI
+- Supports reading translation schemas from azure as well as from disk
+    - Enables leaving the common translation schemas on disk if that's preferred
+- Enables easy rollback
+    - Post MVP, a UI could be enabled to allow selecting what version to go back to
+- Supports reading files from azure or from disk
+
+#### Cons
+- Need to support to different URI schemes
+- Relies on pulling files out of azure which is slower than disk or the database
+- Requires more complicated error handling
+
+## Managing translation schemas
+The translation schemas break down into two categories:
+
+- translation schemas dedicated to a specific sender or receiver
+  - one of them will typically be set in the `setting` for either the sender or receiver
+- translation schemas that are used across multiple senders or receivers across organizations
+    - for now, these will continue to be stored in the source code since changes here should be carefully rolled out
```

Comment:
```suggestion
    - for now, these will continue to be stored in the source code since changes here should be carefully rolled out and they are not subject to frequent changes
```